city_scrapers/spiders/chi_state_university.py

```diff
@@ -0,0 +1,151 @@
+                        location=self._parse_location(item),
+                        links=self._parse_links(item),
+                        source=self._parse_source(response),
+                    )
+                    meeting["status"] = self._get_status(meeting)
+                    meeting["id"] = self._get_id(meeting)
+
+                    yield meeting
+
+    def _parse_description(self, item):
+        parts = item.xpath(".//text()").extract()
+        description = "".join([str(x) for x in parts])
+        description = (
+            description.replace("\xa0", " ")
+            .replace("\n", " ")
+            .replace("                                     ", " ")
+            .replace("\t", " ")
+            .strip()
+        )
+        return description
+
+    def _parse_classification(self, i):
+        classifications = [BOARD, BOARD, COMMITTEE]
+        return classifications[i]
+
+    def _parse_start(self, item):
+        text = item.xpath(".//text()").extract()[0]
+        text = text.replace("\xa0", "").strip()
+
+        today = datetime.now()
+        yearMatch = today.year
+        monthMatch = today.month
+        dayMatch = today.day
+        hourMatch = 0
+        minuteMatch = 0
+
+        try:
+            yearMatch = int(re.search(r"\d{4}", text).group(0))
+        except AttributeError:
+            pass
+
+        try:
+            dayMatch = int(re.search(r"\d{1,2}", text).group(0))
+        except AttributeError:
+            pass
+
+        try:
+            months = {m.lower() for m in month_name[1:]}
+            monthMatch = next(
+                (word for word in text.split() if word.lower() in months), None
+            )
+            monthMatch = datetime.strptime(monthMatch, "%B").month
+        except AttributeError:
+            pass
+
+        try:
+            minuteMatch = re.search(r":([0-5][0-9])", text).group(0)
+            minuteMatch = int(minuteMatch.replace(":", ""))
+        except AttributeError:
+            pass
+
+        try:
+            hourMatch = re.search(r"(1[0-2]|0?[1-9]):", text).group(0)
+            hourMatch = int(hourMatch.replace(":", ""))
+        except AttributeError:
+            pass
+
+        if "p.m." in text:
+            hourMatch += 12
+
+        return datetime(yearMatch, monthMatch, dayMatch, hourMatch, minuteMatch)
+
+    def _parse_time_notes(self, item):
+        text = " ".join(item.xpath(".//text()").extract())
+        text = text.replace("\xa0", "").strip()
+
+        notes = ""
+        if "reschedule" in text.lower() or "postpone" in text.lower():
+            notes = text
+        if "tbd" in text.lower():
+            notes = "Meeting time TBD"
+        return notes
+
+    def _parse_location(self, item):
+
+        return {
+            "address": "9501 S. King Drive Chicago, IL 60628",
+            "name": "Room 15, 4th Floor, Gwendolyn Brooks Library Auditorium",
+        }
+
+    def _parse_links(self, item):
```

Comment:
It looks like they're posting agendas and minutes as well, and we'll want to make sure to parse those https://www.csu.edu/boardoftrustees/meetingagendas/year2020.htm

You can find an example of parsing the documents first and then parsing meetings in [`chi_north_river_mental_health`](https://github.com/City-Bureau/city-scrapers/blob/5c7c9fef9e87ae71bb9c4ba7a6156b6ba1e84883/city_scrapers/spiders/chi_north_river_mental_health.py)

Reply:
In the example you have given, they appear to be parsing minutes separately to normal meetings. 
In my case, would it be reasonable to stick with what I have, then check the minutes url to see if there is an associated link to the minutes and include it?

Reply:
Sorry about that, this is a better example of what I had in mind

https://github.com/City-Bureau/city-scrapers-akr/blob/c451c5b3b90b3f7b0e44201978b9c3e27b44947c/city_scrapers/spiders/summ_library.py#L16-L22

We could first check the minutes page, build an index of dates to documents, and then parse the schedule page attaching documents wherever they match